examples/vpc.nix

```diff
@@ -0,0 +1,126 @@
+{
+  region ? "us-east-1"
+, accessKeyId 
+, ...
+}:
+with (import <nixpkgs> {}).lib;
+{
+  
+  machine =
+    {config, resources, pkgs, lib, ...}:
+    {
+      deployment.targetEnv = "ec2";
+      deployment.ec2.accessKeyId = accessKeyId;
+      deployment.ec2.associatePublicIpAddress = true;
+      deployment.ec2.region = region;
+      deployment.ec2.instanceType = "r3.xlarge";
+      deployment.ec2.subnetId = resources.vpcSubnets.subnet-b;
+      deployment.ec2.keyPair = resources.ec2KeyPairs.keypair.name;
+      deployment.ec2.securityGroups = [];
+      deployment.ec2.securityGroupIds = [ resources.ec2SecurityGroups.sg.name ];
+    };
+
+  resources.ec2KeyPairs.keypair = { inherit region accessKeyId; };
+
+  resources.ec2SecurityGroups = {
+    sg =
+      { resources, lib, ... }:
+      {
+        inherit region accessKeyId;
+        vpcId = resources.vpc.vpc-nixops;
+        rules = [
+          { toPort = 22; fromPort = 22; sourceIp = "41.231.120.171/32"; }
```

Comment:
Please add a check if `fromPort` is set otherwise the following error message is thrown: 

```
AttributeError: 'NoneType' object has no attribute 'get'
```

from


```
from_port = int(rule_xml.find("attr[@name='fromPort']/int").get("value"))
```